### ISSUE TITLE
add an sqp algorithm

### ISSUE THREAD

**Issue by hdakpo** [hdakpo/sfaR#72]
NlcOptim offers an interesting sqp algorithm. I forsee that it can be used for finding starting values for complex optimizations.

Moreover I anticipate that it can be used for constrained optimization (e.g. imposition of monotonicity properties)

**Comment by arne-henningsen** [comment]
For consistency and modularity, I recommend to implement additional optimisation algorithms in the "maxLik" package. If you prefer GitHub over R-Forge, I can move the development of the "maxLik" package from R-Forge to GitHub. I can also help with the implementation of new optimisation algorithms.

**Comment by hdakpo** [comment]
Do you mean here to re-write the optimization algorithms themselves or to just bring what has already been done in maxLik (in the same image as optimx)? For instance if I take the mla algorithm, it has already been optimized  (using c++)in the package marqLevAlg. So what we do is to link the package to maxLik and create a uniform way of the results presentation. This means that the dependencies will go through maxLik....

**Comment by arne-henningsen** [comment]
Yes, the latter. Indeed, most optimization algorithms that are available in maxLik use external packages. I suggest that, e.g., maxLik( ..., method = "sqp" ) uses package "NlcOptim" for the optimisation and returns an object that is consistent with the object returned by maxLik() when using other optimization methods. Then, sfacross() and other functions in the sfaR package can easily use the "sqp" optimisation method (and all other optimisation methods implemented in maxLik).

**Comment by hdakpo** [comment]
this is great. It is a really wonderful idea. Therefore, if you can bring the maxLik package on github, then I will join the project in adding these optimization algorithms. I also see that it will simplify some of the code in the sfaR package.
